salt/pillar/__init__.py

```diff
@@ -642,11 +642,13 @@ def render_pstate(self, sls, saltenv, mods, defaults=None):
                                     errors += err
         return state, mods, errors
 
-    def render_pillar(self, matches):
+    def render_pillar(self, matches, errors=None):
         '''
         Extract the sls pillar files from the matches and render them into the
         pillar
         '''
+        if errors is None:
+            errors = []
         pillar = copy.copy(self.pillar_override)
         errors = []
```

Comment:
You're resetting errors to an empty list here. Looking at the code above, I believe this is not the desired behavior...